force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -124,18 +158,19 @@ public with sharing class ClassesObjectsMethods {
     public static String findTheScientist() {
         // The list of scientists' names
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
-
         // The variable to store the concatenated string
-
+        String finalNames = '';
         // Loop through the list of scientists' names
-
-            // If the name doesn't contain the letter 'G', skip this iteration
-
+        for(String n : scientistNames){
             // Add the name to the result string, followed by a comma
-
+            if(n.toLowerCase().contains('g')){
+                finalNames += n + ',';
+                System.debug(finalNames);
+            }
```

Comment:
good. I would have named my index variable name and not n for the sake of readability.